File: 19_remove_nth_node_from_end_of_list.py
def removeNthFromEnd(head, n):
    if head.next is None:
        return None

    slow = head
    fast = head

    for i in range(0, n):
        fast = fast.next

    while fast is not None and fast.next is not None:
        slow = slow.next
        fast = fast.next

    if fast is None:
        head = slow.next
    else:
        slow.next = slow.next.next

    return head

File: 0_99/19_remove_nth_node_from_end_of_list.go
package main

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	if head.Next == nil {
		return nil
	}

	slow := head
	fast := head

	for i := 0; i < n; i++ {
		fast = fast.Next
	}

	for fast != nil && fast.Next != nil {
		slow = slow.Next
		fast = fast.Next
	}

	if fast == nil {
		head = slow.Next
	} else {
		slow.Next = slow.Next.Next
	}

	return head
}
